pkg/asset/machines/openstack/machines.go

```diff
@@ -37,9 +39,16 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	platform := config.Platform.OpenStack
 
 	az := ""
-	trunk := config.Platform.OpenStack.TrunkSupport
+	trunk := platform.TrunkSupport
+
 	provider := generateProvider(clusterID, platform, pool.Platform.OpenStack, osImage, az, role, userDataSecret, trunk)
 
+	for _, apply := range providerOptions {
```

Comment:
If several providerOptions were passed, does it mean that you will overwrite previous ServerGroupID values? Because in `WithServerGroups` I see `provider.ServerGroupID = sg.ID` on line 172.

Reply:
yes, exactly. Do you think that the behaviour should be different?